app/_hub/kong-inc/openid-connect/how-to/_demonstrating_proof_of_possession.md

```diff
@@ -0,0 +1,148 @@
+
+The following example shows how to enable this feature for the JWT Access Token Authentication method. Similar steps can be followed for the other methods.
+
+## Demo
+### Prerequisites
+
+Follow these prerequisites to set up a demo Keycloak app and a Kong service and route for testing mTLS client auth.
+
+{% include_cached /md/plugins-hub/oidc-prereqs.md %}
+
+### Set up Demonstrating Proof-of-Possession with the OIDC plugin
+
+To enable Demonstrating Proof-of-Possession, use the [`proof_of_possession_dpop`](/hub/kong-inc/openid-connect/configuration/#config-proof_of_possession_dpop) configuration option,
+and set the service name to `openid-connect` to follow the demo:
+
+<!-- vale off -->
+{% plugin_example %}
+plugin: kong-inc/openid-connect
+name: openid-connect
+config:
+  issuer: "http://keycloak.test:8440/auth/realms/master"
+  client_id: "cert_bound"
+  client_secret: "cf4c655a-0622-4ce6-a0de-d3353ef0b714"
+  auth_methods:
+    - "bearer"
+  proof_of_possession_dpop: "strict"
+targets:
+  - service
+formats:
+  - konnect
+  - curl
+  - yaml
+  - kubernetes
+{% endplugin_example %}
+<!-- vale on -->
+
+If this is configured correctly, it returns a `200` response and something like the following:
+
+
+```json
+{
+    "id": "5f35b796-ced6-4c00-9b2a-90eef745f4f9",
+    "name": "openid-connect",
+    "service": {
+        "id": "5fa9e468-0007-4d7e-9aeb-49ca9edd6ccd"
+    },
+    "config": {
+        "issuer": "https://keycloak.test:8440/auth/realms/master",
+        "client_id": [ "cert-bound" ],
+        "client_secret": [ "cf4c655a-0622-4ce6-a0de-d3353ef0b714" ],
+        "auth_methods": [ "bearer" ],
+        "proof_of_possession_dpop": "strict",
+    }
+}
+```
+
+### Test Demonstrating Proof-of-Possession
+
+Obtain the token from the IdP. A DPoP supporting client is required. For this demo, here is an example using [`oauth2c`](https://github.com/cloudentity/oauth2c.git):
+
+```bash
+oauth2c https://keycloak.test:8440/realms/master/.well-known/openid-configuration \
+--client-id cert-bound --client-secret cf4c655a-0622-4ce6-a0de-d3353ef0b714       \
+--grant-type client_credentials --scopes openid --dpop                            \
+--signing-key https://path/to/the_jwks.json
+```
+
+If this is configured correctly, it returns a `200` response and something like the following:
+```json
+{
+    "access_token": "eyJhbG...",
+    "token_type":"DPoP"
+}
+```
+
+The token you obtain should include a claim that consists of the hash of the client certificate:
+```json
+{
+    "exp": 1622556713,
+    "typ": "Bearer",
+    "cnf": {
+        "jkt": "hh_XBS..."
+    }
+}
+```
+
+Access the service using the same client certificate and key used to obtain the token.
+The client should generate proper proof for possession of the key and send it via the `DPoP` header, which will be verified by Kong.
```

Comment:
`oauth2c` cannot be used to initiate an RP request thus we cannot use it in this step. I cannot find a proper 3rd party tool to do the demo here.